spec/labor/loggers/log_worker_queue_stats_spec.rb

```diff
@@ -0,0 +1,15 @@
+require "rails_helper"
+
+describe Loggers::LogWorkerQueueStats do
+  it "logs totals" do
+    allow(described_class).to receive(:record_totals)
```

Comment:
it's a bit weird that we're testing the private methods of a class, this test would break if we change the name of the private methods, which probably shouldn't.

maybe we can `allow/expect` the `DataDogStatsClient` since it's what it's ultimately doing, sending events to Datadog

Reply:
This way we are just testing that _we call the private method_ but nothing actually within it. I think if we reach in and start stubbing and testing what it is in the private method the ability to change it in the future becomes harder and we break that "rule" even more of dont test your private methods.

Reply:
The argument stands for both approaches (mine included), the tests should not, as much as possible, know anything about how a class is implemented, so they should break if we refactor the class internals. 

I though of a different approach which would have been to remove the expectations and record the operation with VCR but the DD client is not enabled in testing.

What if we use dependency injection, which to me is a good idea for a what essentially is a facade/utility class?  

`.run` could take the DD client as an argument which defaults to `DataDogStatsClient` and in the tests we can inject whatever we want. 

This would also allow anyone in the future to reuse the class `LogWorkerQueueStats` to send events to multiple destinations (one could be HoneyComb?) without too much code, you just need a "duck object" that responds to `.gauge`

Reply:
I'm really not a fan of adding an argument to a class just for the purpose of making the tests easier to write when we don't really need an argument in the first place. I would rather stub the `DataDogStatsClient` client than change the actual class structure for the tests.

Reply:
I understand, though I don't totally agree, in the sense that classes should have explicit external dependencies as much as possible in general, but that's an argument for a different kind of conversation or overall code structure

Anyway, let's keep the code as is for now